test/unit/contrib/parallel/reader_writer_latch_test.cpp

```diff
@@ -0,0 +1,98 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <seqan3/contrib/parallel/buffer_queue.hpp>
+#include <seqan3/contrib/parallel/buffer_queue.hpp>
+#include <seqan3/contrib/parallel/reader_writer_latch.hpp>
+
+using namespace seqan3::contrib;
+
+TEST(reader_writer_latch, parallel)
+{
+    size_t threads = std::thread::hardware_concurrency();
+
+    if (threads > 4)
+        threads = 4;
```

Comment:
why do we limit this?

Reply:
In general, in SeqAn we had issues with travis.